torch/onnx/symbolic_opset11.py

```diff
@@ -663,14 +663,16 @@ def flatten(g, input, start_dim, end_dim):
     # if end_dim is negative add dim
     if end_dim < 0 :
         end_dim = dim + end_dim
-    input_dims = input.type().sizes()
-    output_dims = []
-    for i in range(0, dim):
-        if start_dim < i and end_dim >= i:
-            output_dims[start_dim] = output_dims[start_dim] * input_dims[i]
-        else:
-            output_dims.append(input_dims[i])
-    shape = g.op("Constant", value_t=torch.LongTensor(output_dims))
+    input_size = g.op("Shape", input)
```

Comment:
What's the difference between opset9 and opset11? Could we abstract out the common part, as it seems there are a lot of repeated code.

Reply:
I think the code is the same for opset 9 and 11, don't know why. ONNX Flatten between opset 9 and 11 differs only that opset 11 can have negative axes